airflow/models/taskmixin.py

```diff
@@ -15,61 +15,78 @@
 # specific language governing permissions and limitations
```

Comment:
does it make sense to keep the file name `taskmixin.py` ?

Reply:
In a local change (that I separated this out from) I'm adding in a new "TaskLike" mixin.

We could change this from taskmixins to just mixins, but probably need to keep the old file for backcompat :(

Reply:
yeah we should keep the old file for backcompact (though I'd be surprised if users use this class in custom code?)
If we are going to have `mixins.py` it probably means that [skipmixin.py](https://github.com/apache/airflow/blob/main/airflow/models/skipmixin.py) should be moved there too?

Reply:
if we're renaming the class anyway what do we need the old module for 🤔

unless you're aliasing `TaskMixin=DependencyMixin` somewhere for compat?

unrelated but i am a tad skeptical of name TaskLike but was waiting to see what the class looks like before trying to think of any alternatives... i know maybe there's not gonna be a better option.  but ideally we could convey something about the quality of the "likeness"  because e.g. what if we compose with multiple classes that are "task-like" in their own indepedent way....  e.g.  TaskLike1 and TaskLike2

Reply:
> unless you're aliasing TaskMixin=DependencyMixin somewhere for compat?

https://github.com/apache/airflow/pull/20297/files#diff-30230355a7c5c56d9bc0b6025e028d5bd5bc500383093de4bde01f5af522eb47R81-R90

Reply:
> though I'd be surprised if users use this class in custom code?

I too would be, but I am quite passionate about support backcompat if we can.

Reply:
> unrelated but i am a tad skeptical of name TaskLike

Yeah, it's not the _best_ name, espeically as it will probably include TaskGroups. It's "a node in the DAG graph, so either a Task/BaseOperator (mapped or unmapped) or a TaskGroup (again, mapped or unmapped)" -- so it will have `dag` properties, and a few other things like upstream/downstream relations.

More than happy for better name ideas!

Reply:
"DagGraphNode" is an option, but in some ways it means as little as "TaskLike"

Reply:
fwiw `DagGraphNode` seems a lot better to me

Reply:
G in DAG already means Graph so this should be `DAGNode`.

Reply:
Yeah I realised overnight that this is a case of RAS Syndrome (RAS = Redundant Acronym Syndrome 😁 )